lib/repos.js

```diff
@@ -0,0 +1,133 @@
+"use strict";
+
+var async = require("async");
+var github = require("octonode");
+var jsdiff = require("diff");
+var chalk = require("chalk");
+
+var TEXT_PADDING_CHAR = "*";
+var DIFF_HEADER_LINES = 4; // number of lines in a diff output header.
+
+/**
+ * Create a multi-repository client.
+ *
+ * @param {Object} opts       options
+ * @param {object} opts.argv  Command line args
+ * @returns {void}
+ */
+var Repos = module.exports = function (opts) {
+  var argv = opts.argv || {};
+
+  // Authenticate.
+  this._client = null;
+  if (argv.ghUser && argv.ghPass) {
+    this._client = github.client({
+      username: argv.ghUser,
+      password: argv.ghPass
+    });
+  } else if (argv.ghToken) {
+    this._client = github.client(argv.ghToken);
+  } else {
+    throw new Error("Must specify ghUser+ghPass or ghToken");
+  }
+
+  // Member state.
+  this._files = argv.files;
+  this._repos = argv.repos;
+  this._format = argv.format;
+  this._transform = argv.transform;
+};
+
+/**
+ * Format content for display.
+ *
+ * @param   {Array}   data  Array of `{ file, repo, content }` files.
+ * @returns {String}        Display-friendly string.
+ */
+Repos.prototype._toDisplay = function (data) {
+  data = data || [];
+
+  if (this._format === "json") {
+    return JSON.stringify(data, null, 2); // eslint-disable-line no-magic-numbers
+  }
+
+  if (this._format === "text") {
+    return data.map(function (obj) {
+      var repoPath = [obj.repo, obj.file].join("/");
+      var padding = 5; // length of repo path + extra padding.
+      var line = new Array(repoPath.length + padding).join(TEXT_PADDING_CHAR);
+
+      return [
+        line,
+        [TEXT_PADDING_CHAR, repoPath, TEXT_PADDING_CHAR].join(" "),
+        line,
+        obj.content
+      ].join("\n");
+    }).join("\n\n");
+  }
+
+  if (this._format === "diff") {
```

Comment:
What is the use case of the diff output? If it's meant for human reading, then I understand the color output below. If it's going to be used to generate patches to apply to remote repository branches, then we shouldn't add color coding.

Maybe the latter output option is simply a new option called `"patch"`.